tests/filecheck/dialects/emitc/emitc_types.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: XDSL_ROUNDTRIP
+
+// Test EmitC_ArrayType with a 2D shape and f32 element type
+"test.op"() {"array_attr" = !emitc.array<4x2xf32>}: ()->()
+// CHECK: array_attr = !emitc.array<4x2xf32>
+
+// Test EmitC_ArrayType with a 1D shape and i32 element type
+"test.op"() {"array_attr" = !emitc.array<10xi32>}: ()->()
+// CHECK: array_attr = !emitc.array<10xi32>
+
+// Test EmitC_ArrayType with a different 1D shape and f64 element type
+"test.op"() {"array_attr" = !emitc.array<5xf64>}: ()->()
+// CHECK: array_attr = !emitc.array<5xf64>
+
+// Test EmitC_ArrayType with a 3D shape and i1 element type
+"test.op"() {"array_attr" = !emitc.array<3x4x5xi1>}: ()->()
+// CHECK: array_attr = !emitc.array<3x4x5xi1>
```

Comment:
This is fine but I'd recommend something a little less verbose since we'll be adding more things:
```suggestion

// CHECK: f32_2D = !emitc.array<4x2xf32>
// CHECK-SAME: i32_1D = !emitc.array<10xi32>
// CHECK-SAME: f64_1D = !emitc.array<5xf64>
// CHECK-SAME: i1_3D = !emitc.array<3x4x5xi1>
"test.op"() {
  f32_2D = !emitc.array<4x2xf32>,
  i32_1D = !emitc.array<10xi32>,
  f64_1D = !emitc.array<5xf64>,
  i1_3D = !emitc.array<3x4x5xi1>,
}: ()->()
```
Also, MLIR puts the checks before, which I've tried to do in new test files.

Reply:
right, thanks!